tests/ResourceSyncingUsingPolymorphicTest.php

```diff
@@ -0,0 +1,349 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\MorphToMany;
+use Illuminate\Support\Facades\Event;
+use Illuminate\Support\Str;
+use Stancl\JobPipeline\JobPipeline;
+use Stancl\Tenancy\Bootstrappers\DatabaseTenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Syncable;
+use Stancl\Tenancy\Contracts\SyncMaster;
+use Stancl\Tenancy\Database\Concerns\CentralConnection;
+use Stancl\Tenancy\Database\Concerns\ResourceSyncing;
+use Stancl\Tenancy\Database\DatabaseConfig;
+use Stancl\Tenancy\Database\Models\TenantPivot;
+use Stancl\Tenancy\Events\SyncedResourceSaved;
+use Stancl\Tenancy\Events\TenancyEnded;
+use Stancl\Tenancy\Events\TenancyInitialized;
+use Stancl\Tenancy\Events\TenantCreated;
+use Stancl\Tenancy\Jobs\CreateDatabase;
+use Stancl\Tenancy\Listeners\BootstrapTenancy;
+use Stancl\Tenancy\Listeners\RevertToCentralContext;
+use Stancl\Tenancy\Listeners\UpdateSyncedResource;
+use Stancl\Tenancy\Tests\Etc\Tenant;
+
+beforeEach(function () {
+    config(['tenancy.bootstrappers' => [
+        DatabaseTenancyBootstrapper::class,
+    ],
+        'tenancy.tenant_model' => ResourceTenantUsingPolymorphic::class,
+    ]);
```

Comment:
Can you:
1. Reformat this to be a normal array
2. Update the config key? https://github.com/archtechx/tenancy/blob/master/assets/config.php#L13

It's strange that the tests are passing even though the model is not actually set correctly.

Are you using `$tenant->users` or `$tenant->companies` anywhere? It seems that the model isn't actually used here.

We should be making assertions that we can access the right users & companies *from* the tenant.

Reply:
> It's strange that the tests are passing even though the model is not actually set correctly.

Tests are passing because the config key is correct. In this PR, config has `tenancy.tenant_model`. The link you provide is in the master. So, I'm going to merge the master branch into this branch, and then we'll see if tests pass with same code.

Reply:
> Are you using $tenant->users or $tenant->companies anywhere? It seems that the model isn't actually used here.

You are right. `$tenant->companies` is not being used anywhere. Tests are passing because TestCase.php set the `Tenant` model.

Reply:
Added the assertions and a test for accessing the right resources from the Tenant model.